app/fetchers/buildpack_lifecycle_fetcher.rb

```diff
@@ -4,14 +4,19 @@ class << self
       def fetch(buildpack_names, stack_name)
         {
           stack: Stack.find(name: stack_name),
-          buildpack_infos: ordered_buildpacks(buildpack_names),
+          buildpack_infos: ordered_buildpacks(buildpack_names, stack_name),
         }
       end
 
       private
 
-      def ordered_buildpacks(buildpack_names)
-        buildpacks = Buildpack.where(name: buildpack_names).all
+      ## NOTE: if a requested system buildpack is not on the requested stack,
+      ##       the BuildpackInfo object will have a name and not a record (even
+      ##       though the buildpack exists). At this point the error returned
+      ##       to the user will probably be VERY confusing
```

Comment:
We currently have cross teams planned with Routing and CLI, so it will be difficult to start another one in the near term.

Reply:
The message was clear to me. I find tests to be valuable in terms of documenting desired/known behavior, in this case documenting what error users will see. Leaving comments in the code that describe behavior is a last resort. Main motivation being that tests go red when things change, whereas many a comment has become obsolete silently. Tests do a good job of showing rather than telling. Lastly, this particular case feels straightforward to set up, but correct me if I'm wrong.

Reply:
So, we just gave this a test, and the error we see is:
```
Staging app and tracing logs...
Unexpected Response
Response code: 500
CC code:       0
CC error code: 
Request ID:    24dddc03-c91c-4fb8-4783-ac3b8568e49c::4ab4e09a-c73d-4466-a3e9-e5daffccf25e
Description:   {
  "description": "Stager error: no compiler defined for requested stack",
  "error_code": "CF-StagerError",
  "code": 170011
}
```

Coming back to comments made earlier by @dgodd : this error doesn't really inform the user well when they have specified a buildpack name, but there is no buildpack with that name for the relevant stack (either specified, or defaulted). 

As outsiders to this code base it appears to us that this error handling is interwoven with the lifecycle, and as such we would really want a cross team to improve this. I suppose this could be acceptable in a first cut, and we could attack this as an improvement later on.

Reply:
@zrob Do you have thoughts on how this error message could be improved? Is it acceptable as is for an MVP?

-- @selzoc and maryam

Reply:
@Gerg ^^